### DISCUSSION TITLE
locale and personal keybinds

### DISCUSSION THREAD

**Discussion by philuser-tech** [nwg-piotr/nwg-shell#340] [Ideas]
I've been having problems with the shortcut keys on a French keyboard on my HP laptop for a few days now. When switching between workspaces, they no longer work. The “superkey” key still works for calls to the general menu, but the SuperKey+2 or SuperKey+Shift+2 associations no longer work. I've checked my files in the `~/.config/hypr/` directory. I had not made any prior changes to the `~/.config/hypr/hyprland.conf` file, where the `# SWITCH WORKSPACES with mainMod + [0-9]` are located. 
To restore normal operation I had to modify the `~/.config/hypr/hyprland.conf` file by commenting out all the `# SWITCH WORKSPACES with mainMod + [0-9]` and the `# MOVE ACTIVE WINDOW TO A WORKSPACE with mainMod + SHIFT + [0-9]` and add a new `source = ~/.config/hypr//UserConfigs/UserKeybinds.conf` file containing :
Of course, this only applies to laptops with a French keyboard. 
`
# Change workspace with mainMod + [0-9]
bind = $mainMod, ampersand,     workspace, 1
bind = $mainMod, eacute,        workspace, 2
bind = $mainMod, quotedbl,      workspace, 3
bind = $mainMod, apostrophe,    workspace, 4
bind = $mainMod, parenleft,     workspace, 5
bind = $mainMod, minus,         workspace, 6
bind = $mainMod, egrave,        workspace, 7
bind = $mainMod, underscore,    workspace, 8
bind = $mainMod, ccedilla,      workspace, 9
bind = $mainMod, agrave,        workspace, 10

# Move active window and tracking to workspace
bind = $mainMod SHIFT, ampersand,   movetoworkspace, 1
bind = $mainMod SHIFT, eacute,      movetoworkspace, 2
bind = $mainMod SHIFT, quotedbl,    movetoworkspace, 3
bind = $mainMod SHIFT, apostrophe,  movetoworkspace, 4
bind = $mainMod SHIFT, parenleft,   movetoworkspace, 5
bind = $mainMod SHIFT, minus,       movetoworkspace, 6
bind = $mainMod SHIFT, egrave,      movetoworkspace, 7
bind = $mainMod SHIFT, underscore,  movetoworkspace, 8
bind = $mainMod SHIFT, ccedilla,    movetoworkspace, 9
bind = $mainMod SHIFT, agrave,      movetoworkspace, 10

# Silently move active window to workspac
bind = $mainMod CTRL, ampersand, movetoworkspacesilent, 1
bind = $mainMod CTRL, eacute, movetoworkspacesilent, 2
bind = $mainMod CTRL, quotedbl, movetoworkspacesilent, 3
bind = $mainMod CTRL, apostrophe, movetoworkspacesilent, 4
bind = $mainMod CTRL, parenleft, movetoworkspacesilent, 5
bind = $mainMod CTRL, minus, movetoworkspacesilent, 6
bind = $mainMod CTRL, egrave, movetoworkspacesilent, 7
bind = $mainMod CTRL, underscore, movetoworkspacesilent, 8
bind = $mainMod CTRL, ccedilla, movetoworkspacesilent, 9
bind = $mainMod CTRL, agrave, movetoworkspacesilent, 10
`
Wouldn't it be a good idea to ask the installer to associate a dedicated configuration file with the type of keyboard used, and to put this configuration in a customization file?

**Comment by nwg-piotr** [comment]
Well, this part of config belongs to the user, at least for now. I have no idea how many various configs we would need for all the possible keyboard layouts.
